src/Illuminate/View/Compilers/Concerns/CompilesEchos.php

```diff
@@ -102,10 +127,23 @@ protected function compileEscapedEchos($value)
      * @param  string  $value
      * @return string
      */
-    protected function applyEchoHandlerFor($value)
+    protected function wrapInEchoHandler($value)
     {
-        return empty($this->echoHandlers)
-            ? $value
-            : "is_object($value) && isset(app('blade.compiler')->echoHandlers[get_class($value)]) ? call_user_func_array(app('blade.compiler')->echoHandlers[get_class($value)], [$value]) : $value";
+        return "app('blade.compiler')->applyEchoHandler({$value})";
+    }
+
+    /**
+     * Apply the echo handler for the value if it exists.
+     *
+     * @param  $value  string
+     * @return string
+     */
+    public function applyEchoHandler($value)
+    {
+        if (is_object($value) && isset($this->echoHandlers[get_class($value)])) {
+            return call_user_func(app('blade.compiler')->echoHandlers[get_class($value)], $value);
```

Comment:
As this is now inside the class, can't we just use `$this` instead of `app('blade.compiler')` here?

Reply:
:+1: